lib/app.rb

```diff
@@ -16,11 +16,11 @@ def build_board
       @simulator.setup_board(board_rows_input, baord_columns_input)
 
       @display.build_board_complete_message
-      @simulator.board.print_board
+      @display.print_board(@simulator.board.board_with_axis)
```

Comment:
you have an `attr_reader` on `@dispaly` why not use it?

also why does `display` need to know how to `print_board` is it the display's responsibility to know anything about a board?

as discussed, we went down the path of moving all the `puts` into a separate domain `Display` so in future we could switch out to another **type** of display like HTML or curses - this helps achieve this

**but** we could also think of another scenario that we should be able to reuse our `Display` class in another game, say an adventure game or a maze game - there is no idea of `print_board` in those cases but maybe more generic `draw_background` or maybe even `draw_screen` ?

what are the core responsibilities of `Display` that would allow us to reuse it in another game?